### PR TITLE
Remove Grisu for Ryu;

### PR DESCRIPTION
**Pull request by isentropic**
There have been a few modification in tests too, details explained in the review

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -36,8 +28,9 @@ end
     @test Showoff.format_fixed_scientific(-Inf, 1, false) == "-∞"
     @test Showoff.format_fixed_scientific(NaN, 1, false) == "NaN"
     @test Showoff.format_fixed_scientific(0.012345678, 4, true) == "12.34568×10⁻³"
-    @test Showoff.format_fixed_scientific(0.012345678, 4, false) == "1.234568×10⁻²"
-    @test Showoff.format_fixed_scientific(-10.0, 4, false) == "-1.000×10¹"
+    @test Showoff.format_fixed_scientific(0.012345678, 4, false) == "1.2346×10⁻²"
```
Comment (isentropic) [inline test/runtests.jl:31]: This test is modified as in the function docstring it says that (line 140 in the original code) the function should match `@printf`.

File: test/runtests.jl [hunk 2]
```diff
@@ -36,8 +28,9 @@ end
     @test Showoff.format_fixed_scientific(-Inf, 1, false) == "-∞"
     @test Showoff.format_fixed_scientific(NaN, 1, false) == "NaN"
     @test Showoff.format_fixed_scientific(0.012345678, 4, true) == "12.34568×10⁻³"
```
Comment (isentropic) [inline test/runtests.jl:30]: This test is failing, as I have still not implemented the engineering flag. Once the rest is approved, I will make sure this test passes, too.

File: test/runtests.jl [hunk 3]
```diff
@@ -36,8 +28,9 @@ end
...
-    @test Showoff.format_fixed_scientific(0.012345678, 4, false) == "1.234568×10⁻²"
-    @test Showoff.format_fixed_scientific(-10.0, 4, false) == "-1.000×10¹"
+    @test Showoff.format_fixed_scientific(0.012345678, 4, false) == "1.2346×10⁻²"
+    @test Showoff.format_fixed_scientific(-10.0, 4, false) == "-1.0000×10¹"
+    @test Showoff.format_fixed_scientific(-2.99999999999999956E-16, 2, false) == "-3.00×10⁻¹⁶"
```
Comment (isentropic) [inline test/runtests.jl:33]: This is a new test, the loingstanding issue https://github.com/JuliaGraphics/Showoff.jl/issues/22

**Review by isentropic — COMMENTED** [review]
Currently passing all the test including the new one, except for engineering notation (for now)

**Comment by timholy** [comment]
I just pushed a branch, `teh/grisu`, which has the work I started on this. Importantly, Grisu is only available on Julia 1.6 and higher, so unless you plan to drop support for earlier Julia versions we need to keep both implementations around.

**Comment by timholy** [comment]
I did mention the existence of that branch in https://github.com/JuliaGraphics/Showoff.jl/issues/28#issuecomment-759253521

**Comment by isentropic** [comment]
Right I just realized it. I'll see if I can get it to work. Surprisingly the code you wrote is almost the same I have here.

**Comment by timholy** [comment]
Indeed! The hard part is supporting the engineering notation; I tried to come up with a cheap trick, like multiplying or dividing by 10 to get it to the nearest multiple of 3, but then you end up doing carries on the `Char`s of the strings, and that's really ugly. So I quit working on it :smile:

**Pull request closed without merging**